proc_IR.py

```diff
@@ -1,77 +1,273 @@
-        ph0 = zeroth_order_ph(ph0, fl=fl)
+        ph0 = zeroth_order_ph(ph0)
         logger.info(strm('phasing dimension as one'))
     else:
         logger.info(strm("there is only one dimension left -- standard 1D zeroth order phasing"))
         ph0 = ph0/abs(ph0)
     s /= ph0
-    fl.next('frequency domain -- after hermitian function test and phasing')
+    fl.next('phased data frequency domain')
     s.ft('t2')
-    fl.image(s.C.convolve('t2',10))
+    fl.image(s.C.convolve('t2',10).setaxis('vd','#').set_units('vd','scan #'))
+    fl.next('phased data freq domain')
+    fl.image(s.C.setaxis(
+'vd','#').set_units('vd','scan #'))
     #}}}
-    #{{{select t2 axis range and 
+    #{{{slicing FID not sure if needed
     s.ift('t2')
     s = s['t2':(0,None)]
     s['t2',0] *= 0.5
+    fl.next('phased and FID sliced')
+    fl.image(s.C.setaxis('vd','#').set_units('vd','scan #'))
+    fl.next('phased and FID sliced -- frequency domain')
     s.ft('t2')
+    fl.image(s.C.setaxis('vd','#').set_units('vd','scan #'))
+    #fl.show();quit()
+    fl.next('signal vs vd')
+    s_sliced = s['ph1',0]['ph2',1]
+    s_sliced.sum('t2')
+    fl.plot(s_sliced,'o')
+    fl.plot(s_sliced.imag,'o')
+    #}}}
+
+    s.ift('t2')
+    fl.next('t domain for apod')
+    fl.image(s.C.setaxis('vd','#').set_units('vd','scan #'))
+    #{{{time shift and apodization
+    t0 = s['ph2',ph2_val]['ph1',ph1_val].C.mean_all_but('t2').argmax('t2').item()
+    s.setaxis('t2', lambda x: x - t0)
+    apod = s.fromaxis('t2') * 0
+    apod['t2':(None,30e-3)] = apod['t2':(None,30e-3)].run(lambda x: tukey(len(x)))
+    s *= apod
     #}}}
-    #{{{recovery curve and fitting
-    s_sliced = s['ph2',coh_sel['ph2']]['ph1',coh_sel['ph1']]*-1 # bc inverted at high powers
-    # below, f_range needs to be defined
-    M0,Mi,R1,vd = sympy.symbols("M_0 M_inf R_1 indirect",real=True)
-    f,T1,g = recovery(s_sliced, (-100,100),
-            guess={M0:-500, Mi:500, R1:1})
-    fl.plot_curve(f,'inversion recovery curve',guess=g)
-fl.show()
+    if postproc == 'spincore_IR_v1':
+        #{{{subplot for imaging before alignment
+        s.ft('t2')
+        s.reorder('ph1',first=True)
+        s.reorder('ph2',first=True)
+        s.reorder('t2',first=False)
+        fl.next('freq domain - before corr, overview')
+        fl.image(s.C.setaxis(
+    'vd','#').set_units('vd','scan #'))
+        fig,(ax1,ax2) = plt.subplots(1,2)
+        fl.next('filtered and apodized -- before alignment', fig=fig)
+        fl.image(s['ph2',ph2_val]['ph1',ph1_val].C.setaxis(
+    'vd','#').set_units('vd','scan #'),ax=ax1)
+        fl.image(abs(s)['ph2',ph2_val]['ph1',ph1_val].C.setaxis(
+    'vd','#').set_units('vd','scan #'),ax=ax2)
+        #}}}
+        #{{{ pre-alignment
+        s_final = s.C
+        s_before = s['vd',:2]
+        s_after = s['vd',3:]
+        s_after.ift(['ph2','ph1'])
+        s_after.ift('t2')
+        s_after.ft('t2')
+        fl.next('before alignment - FT')
```

Comment:
I don't understand why this whole block is spincore-related.  The *only* thing that I've seen that should be spincore specific is the clock correction.